src/panels/alarm-table/module.js

```diff
@@ -687,23 +687,34 @@ class AlarmTableCtrl extends MetricsPanelCtrl {
 
   performAlarmActionOnDatasource(source, action, alarmId) {
     let self = this;
-    this.datasourceSrv.get(source).then(ds => {
-      if (ds.type && ds.type.indexOf("entity-datasource") < 0) {
-        throw {message: 'Only OpenNMS datasources are supported'};
-      } else {
-        if (!ds[action]) {
-          throw {message: 'Action ' + action + ' not implemented by datasource ' + ds.name + " of type " + ds.type};
+    
+    return new Promise((resolve, reject) => {
```

Comment:
Is there a reason you're making a new `Promise` when `this.datasourceSrv.get()` already starts a promise chain?  You could just return it...

Reply:
Yes `performAlarmActionOnDatasource` wasn't returning a promise necessarily (in other conditions), making .then() undefined for `performAlarmActionOnDatasource`.. so wrapping it in a new promise helped in creating the chain..

Reply:
Ah, ok, thanks.